### PR TITLE
rename enable var to correct name and only use it when needed

### PR DESCRIPTION
**Pull request by mikedanese**
See https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/config-default.sh#L105

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -937,8 +937,6 @@ case ${JOB_NAME} in
     : ${E2E_TEST:="false"}
     : ${E2E_DOWN:="false"}
     : ${NUM_MINIONS:=5}
-    : ${ENABLE_DEPLOYMENTS:=true}
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: I assume these is removed because the current latest release doesn't support them?

Comment (mikedanese) [inline hack/jenkins/e2e.sh]: Yup.

**Comment by ihmccreery** [comment]
LGTM.

**Comment by mikedanese** [comment]
Fixin tests for v1.1 :)

**Comment by justinsb** [comment]
@mikedanese I'm hoping you can explain what's going on here, as I'm trying to figure it out for #20065.  AFAICT, we set the variable ENABLE_DEPLOYMENTS as a shell-scoped variable, and then we export it as KUBE_ENABLE_DEPLOYMENTS:

https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/e2e.sh#L920
```
export KUBE_ENABLE_DEPLOYMENTS=${ENABLE_DEPLOYMENTS:-}
```

However, we do then `source` the script we eventually execute (e2e-runner.sh).  But:

1. I don't know why bother exporting the variables (unless it's so that e2e-runner.sh can execute things)
1. I don't know why we have different names, instead of just keeping it as KUBE_ENABLE_DEPLOYMENTS throughout (unless we want to be clear that we aren't exporting it yet?)

**Pull request merged**